### PR TITLE
egressgw: init gatewayIP to 0.0.0.0

### PR DESCRIPTION
**Pull request by MrFreezeex**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/master/USERS.md)
- [x] Thanks for contributing!

Always init gatewayIP to 0.0.0.0 by default instead of the previous nil value. Before this commit the rules that didn't match any node where added in addMissingEgressRules and removed right after in removeUnusedEgressRules. The egressmap auto convert nil to 0.0.0.0 and removeUnusedEgressRules doesn't do anything to match nil and 0.0.0.0.

```release-note
Prevent egress gateway from adding and then immediately removing BPF policy entries for policies that don't match any gateway node 
```

### CODE REVIEW

File: pkg/egressgateway/policy.go [hunk 1]
```diff
@@ -104,6 +104,7 @@ func (config *PolicyConfig) regenerateGatewayConfig(manager *Manager) {
 	}
 
 	policyGwc := config.policyGwConfig
+	gwc.gatewayIP = zeroIPv4
```
Comment (jibi) [inline pkg/egressgateway/policy.go]: could you move this inside 

```go
gwc := gatewayConfig{
		egressIP: net.IPNet{IP: net.IPv4zero, Mask: net.CIDRMask(0, 0)},
	}
```

please?

Comment (MrFreezeex) [reply pkg/egressgateway/policy.go]: Ah yep indeed that's better thanks for the suggestion :+1:

**Comment by jibi** [comment]
/test-1.26-net-next

**Comment by jibi** [comment]
/test-1.26-net-next

**Comment by jibi** [comment]
(looks like net-next is hitting https://github.com/cilium/cilium/issues/24667)

**Comment by jibi** [comment]
`net-next` failure looks like an unrelated flake, marking as ready

**Comment by pchaigno** [comment]
@jibi Note I don't think MLH managed to create the flake issue.

**Comment by pchaigno** [comment]
We're also missing an approving review.

**Review by jibi — APPROVED** [review]

**Comment by jibi** [comment]
> We're also missing an approving review.

I thought I had already approved it

**Pull request merged**